### ISSUE TITLE
[1.20.1 Fabric 1.8.2] Underwater music plays on land

### ISSUE THREAD

**Issue by 0DeltaWaves** [sf-inc/music_control#63]
Underwater music (Axolotl, Shuniji, Dragonfish) keeps playing on land even though nothing in my music settings suggests this should happen. Any ideas?

![settings 1](https://i.imgur.com/iGY6wLw.png)
![settings 2](https://i.imgur.com/ttexBVF.png)
![settings 3](https://i.imgur.com/N5e3IhX.png)

**Comment by charlyb01** [comment]
Hi, thanks for the report. Are you playing with other music mods? I don't see anything that could cause this issue, except another mod. 

Where are you when this happens exactly?

**Comment by 0DeltaWaves** [comment]
Sorry, I should've included the mod list + log. [Here it is](https://mclo.gs/nbHVKWw)
I don't see any entries when the music starts playing, but I'm including it in case there's anything else that might be a clue. No other music mods.
It's been happening randomly, mostly in a plains biome.

**Comment by charlyb01** [comment]
Do you have any event that contains the "Under Water" event? In this screenshot for example, you can see that it doesn't contain any event. However, you can't see if it's included in others, so you need to check manually if other events contain it. You can also check it the resource pack json with Ctrl+F, that might be easier `resourcepacks\music_control\assets\minecraft\sounds.json`
![image](https://github.com/user-attachments/assets/babb376b-3f64-4f40-80a4-4ea75654cbc8)

**Comment by 0DeltaWaves** [comment]
It's not included in any other event, unfortunately...
![ss](https://i.imgur.com/JZG6UIO.png)

**Comment by charlyb01** [comment]
I acknowledge my explanation with the screenshot is a bit clunky. Your screenshot shows the event that "under water" contains, not the events containing "under water". There is no way in the mod to know by which events one is contained, other than checking all other events. That's why checking the resource pack file might be an easier check. 

However, if you didn't touch these before, there is no chance it's included in another event... I wonder what could cause the issue, I'll try to investigate

**Comment by 0DeltaWaves** [comment]
I'd like to add that it's also not playing creative music while in creative mode (it just plays survival music). Same thing happening to my partner as well, and the issue continues after we both uninstalled + reinstalled the mod and reset all its files. I'm not sure what the issue could be. Let me know if there's any further info I could provide that might help.

**Comment by 0DeltaWaves** [comment]
https://github.com/user-attachments/assets/9749bd31-a315-4d69-86b5-85e070366a19

When I test the creative music in the play tab, it starts playing a survival track, despite the creative event track list only containing creative tracks

**Comment by charlyb01** [comment]
The creative event is played differently with this mod. You can set when it should play in the config.

I think that when you play it in creative, it's playing creative music. The creative event contains creative music + the Overworld event by default (this is vanilla)

**Comment by 0DeltaWaves** [comment]
Is it "creative event fallback"? I changed this to no, but it's still playing exclusively survival music when I'm in creative mode. I'm not really seeing anything else that could be related in the config.

**Comment by charlyb01** [comment]
If you set it to No, then you have the vanilla behavior of the creative event always playing in creative. As I told you, the Creative event contains creative music, and the "game" event music too, meaning the creative event can "play the game event" (which refers to the overworld event in my mod). You can remove it and tell me if you still have other music playing or not

![image](https://github.com/user-attachments/assets/89949b35-cc36-4bef-8a01-49573c5e1384)

**Comment by 0DeltaWaves** [comment]
Yes, I've changed the config option and have had the overworld event removed from the creative event in-game, and it is still playing survival tracks only. The only way either of us have found to fix this and the underwater music playing on land is to uninstall the mod unfortunately

**Comment by charlyb01** [comment]
The only thing I see would be a mod incompatibility for the creative thing. The underwater one is also fixed by removing my mod? I would guess you have a mod adding a new music type, instead of events, which would break my mixin there. If you could test with less mods, that would help a lot. Or if by chance you know what mod added this line to your log `Play Music: minecraft:music.menu{newSystem false, range 16.000000}`. This is the only lead I have atm
